cli/src/commands/squash.rs

```diff
@@ -223,6 +250,18 @@ pub(crate) fn cmd_squash(
         };
         commit_builder.set_description(new_description);
         commit_builder.write(tx.repo_mut())?;
+
+        if args.restore_descendants {
+            // If !args.restore_descendants, the corresponding steps are done inside
+            // tx.finish()
+            let num_reparented = tx.repo_mut().reparent_descendants()?;
+            if let Some(mut formatter) = ui.status_formatter() {
+                writeln!(
+                    formatter,
+                    "Rebased {num_reparented} descendant commits (while preserving their content)",
+                )?;
+            }
+        }
```

Comment:
Can't we simply omit reparenting/rebasing in this case? There should be no A-B+A problem.

(I haven't read the last commit, so feel free to ignore my comment if the reason was described in it.)